### PR TITLE
Tv4g2 issue2036 tripaldbx extension not installed

### LINKED ISSUE #2036: tripaldbx extension will not be installed after import
### Tripal Version

4.0.alpha3

### Drupal Version

10.3.1 

### PostgreSQL Version

16.2 

### PHP Version

8.3.10

### General Description

When trying to import a configuration using Drupal's [Configuration Manager](https://www.drupal.org/docs/administering-a-drupal-site/configuration-management/managing-your-sites-configuration), after exporting a config using the same mechanism, an error is reported:

`Configuration tripaldbx.settings depends on the tripaldbx extension that will not be installed after import.`

This may be caused by artifacts from when tripaldbx was its own module but no longer is. The initial error indicates that the tripaldbx module will not be installed after the import because it is indeed not in the core.extensions.yml file included in the config export (this file lists all of the dependencies required for a successful import). Adding an entry manually for tripaldbx in this file and trying to import (see below for steps) results in the following error:

`Unable to install the tripaldbx module since it does not exist.`

### Steps to reproduce

1. Export a full site configuration using the Configuration Manager: `/admin/config/development/configuration/full/export`
2. Optionally make changes in the configuration (not required).
3. Re-archive the yml files:
    ```shell
    tar -czvf my_config.tar.gz *.yml
    ```
4. Import the config archive: `/admin/config/development/configuration/full/import`
5. Optionally compare changes (if any)
6. Click the "Import all" button: `/admin/config/development/configuration`
7. Get the error message about tripaldbx.settings
8. Optionally clean up your config sync directory (found in the sites/default/files directory, starts with `config_` and has a long string of random characters):
    ```shell
    rm sites/default/files/config_*/sync/*.yml
    ```

### Error messages

```shell
Configuration tripaldbx.settings depends on the tripaldbx extension that will not be installed after import.
```

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Ferrisx4**
# Bug Fix

## Description

This change removes all traces of `tripaldbx.settings` by moving the respective config lines into Tripal itself.

This change is necessary because it prevents sites from being able to import full site configurations that had been generated from sites that still had `tripaldbx.settings` records. 

## Testing?

1. Build a completely new Docker with this branch†
2. Export a full site configuration using the Configuration Manager: `/admin/config/development/configuration/full/export`
3. Make a simple change in the site's configuration, such as rearranging the fields on the Organism content type: `/admin/structure/bio_data/manage/organism/display`
5. Re-import the config archive: `/admin/config/development/configuration/full/import`
6. Do **not** get the following error message, (or any error message 🤞): `Configuration tripaldbx.settings depends on the tripaldbx extension that will not be installed after import.`, and apply the changes.
7. Navigate back to `/admin/structure/bio_data/manage/organism/display` to see that your changes made after the export (step 2) were reverted.
    

† @dsenalik had an issue where Docker kept files around that had been deleted during the build process. His workaround was to [use empty placeholder files](https://github.com/tripal/tripal/issues/2036#issuecomment-2494607062). If your Docker builds and the two deleted files remain, you'll also have to use placeholders.

### CODE REVIEW

**Review by dsenalik — CHANGES_REQUESTED** [review]
1. We need to change remaining `tripaldbx.settings` in the tests, this command will do it: `sed -i -e 's/tripaldbx.settings/tripal.settings/' tripal/tests/src/Unit/TripalDbx/TripalDbxUnitTest.php tripal/tests/src/Kernel/TripalDBX/ConnectionTest.php tripal/tests/src/Kernel/TripalDBX/SchemaTest.php tripal/tests/src/Kernel/TripalDBX/TripalDbxFunctionalTest.php tripal_chado/tests/src/Traits/ChadoTestTrait.php tripal_chado/tests/src/Kernel/ChadoTestKernelBase.php`
2. I think we should have the two placeholder files in this PR - they can be removed later after nightly docker builds - that way nobody needs to do anything special. I have another obsolete file from publish that should be remove, I can make a PR to do it later.
`echo '#' > tripal/config/install/tripaldbx.settings.yml ; echo '#' > tripal/config/schema/tripaldbx.schema.yml`
3. Sadly, I think we need an update hook for existing sites

Running tests locally with items 1 and 2 changed
```
PHPUnit 10.5.38 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.3.13
Configuration: /var/www/drupal/web/modules/contrib/tripal/phpunit.xml

...............................................................  63 / 262 ( 24%)
....III..................I...............IIII................I. 126 / 262 ( 48%)
..............................................................F 189 / 262 ( 72%)
............................................................... 252 / 262 ( 96%)
...I......                                                      262 / 262 (100%)

Time: 01:26:59.986, Memory: 135.45 MB
```
This failing test is unrelated, ~~will be fixed by PR #2039 😆~~ an artifact of my home internet 🙁

File: tripal/config/install/tripal.settings.yml [hunk 1]
```diff
@@ -14,3 +14,19 @@ files:
   quota: 1000000000
   # Number of days until files are automatically deleted (defaults to 60 days)
   expiration: 60
+
+# 'reserved_schema_patterns' must be an associative array where keys are
+# schema name patterns to reserve and values are the reservation descriptions.
+# If you need wildcards, you can use '*'. It is also possible to use regex
+# without delimitation characters (ie. starting and indening '/').
```
Comment (dsenalik) [inline tripal/config/install/tripal.settings.yml]:
```suggestion
# without delimitation characters (ie. starting and ending '/').
```

**Comment by dsenalik** [comment]
I think the update hook is going to be necessary.
I build a docker on 4.x, then switched to this branch. Import failed, even after cache rebuild
8036 4 x

**Comment by Ferrisx4** [comment]
Thanks @dsenalik for the sed command :) Changes have been applied in https://github.com/tripal/tripal/pull/2037/commits/d6d3c085b26372451165079d18bcd3a71655f64a.

I'll work on an update hook to hopefully close out this issue.

File: tripal/tripal.install [hunk 2]
```diff
@@ -1034,3 +1034,18 @@ function tripal_update_10401() {
     throw new DrupalUpdateException('Could not add the tripal_pub_library_query table:' . $error);
   }
 }
+
+/**
+ * Implements hook_update_N().
+ ()
+ * Removes tripaldbx.settings and 
+ */
```
Comment (dsenalik) [inline tripal/tripal.install]:
```suggestion
/**
 * Removes tripaldbx.settings, these have been moved to tripal.settings
 */
```

**Review by dsenalik — CHANGES_REQUESTED** [review]
I built a docker on 4.x
Switched to this branch
Ran the update hook
Exported configuration
Switched order of genus and species on organism
Imported configuration
No errors, and the order was correctly reverted to the original

Only one change needed, the header for the update hook wasn't finished. With suggested change you will see
```
root@2732b8cc8989:/var/www/drupal# drush updatedb
 -------- ----------- --------------- ------------------------------------- 
  Module   Update ID   Type            Description                          
 -------- ----------- --------------- ------------------------------------- 
  tripal   10402       hook_update_n   10402 - Removes tripaldbx.settings,  
                                       these have been moved to             
                                       tripal.settings                      
 -------- ----------- --------------- ------------------------------------- 


 ┌ Do you wish to run the specified pending updates? ───────────┐
 │ ● Yes / ○ No                                                 │
 └──────────────────────────────────────────────────────────────┘
```

File: tripal/tripal.install (hunk 2 above)
Comment (Ferrisx4) [reply tripal/tripal.install]: Good catch! I'm fine with that change.

**Review by dsenalik — APPROVED** [review]
Looks good!

**Pull request merged**